### PR TITLE
Add logic for external SSL cert

### LINKED ISSUE #17: Add additional security - ability to define your own SSL certificate
These tasks were split out from #9, and include:

- Secure Nexus repo with credential
- Turn on CCM Client & Service Salts
- Generate `Register-InternetEndpoint.ps1` script to pair with a `ClientSetup.ps1` script

### PR DESCRIPTION
**Pull request by adilio**
Closes #17 .

Summary of changes:

- Update Set-SSLSecurity.ps1 with logic to:
  - Accept external cert
  - Copy to appropriate cert store
  - Reassign services and bindings
  - Import cert into Java Key Store for Nexus
  - Update firewall rule for Nexus to 8443
  - Generate a `Register-C4bEndpoint.ps1` script to ease on-boarding of clients
  - Updated URI's in `ClientSetup.ps1` and `ChocolateyInstall.ps1` to reflect cert CN/Subject
  - Added JSON values to pass to Jenkins script
- Update Jenkins job templates in the `jenkins` folder to reflect change of port for repo URI
- Move order of Jenkins script to the end again, as we need the CN/subject to update repo URI
- Moved all functions into `Get-Functions.ps1` script, to clean up code and allow cross-script usage

### CODE REVIEW

File: Set-SSLSecurity.ps1 [hunk 1]
```diff
@@ -68,6 +68,9 @@ process {
     # Generate Nexus keystore
     New-NexusCert -Thumbprint $Certificate.Thumbprint
 
+    # Add firewall rule for Nexus
+    netsh advfirewall firewall add rule name="Nexus-8443" dir=in action=allow protocol=tcp localport=8443
```
Comment (steviecoaster) [inline Set-SSLSecurity.ps1:72]: I love CoPilot

**Review by steviecoaster — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -23,31 +23,30 @@ Let's get started!
...
-1. **Automation Pipeline**: These are the workstation or server endpoints you wish to manage packages on, with Chocolatey. Every node requires a license.
 
-Repo Options:
-- Jfrog [Artifactory](https://jfrog.com/artifactory/)
-- Inedo ProGet
-- Other NuGet V2 options discussed here: [Repository Options](xref:host-packages)
+1. **Chocolatey Central Management (CCM) Server App**: CCM is the Web UI portal for your entire Chocolatey environment. Your endpoints check-in to CCM to report their package status. This includes the Chocolatey packages they have installed, and whther any of these packages are outdated. And now, with CCM Deployments, you can also deploy packages or package updates to groups of endpoints, as well as ad-hoc PowerShell commands. CCM is backed by an MS SQL Database. This guide will set up MS SQL Express for you.
```
[the remark this answers is not in the document]
Comment (steviecoaster) [reply README.md:38]: I'm not seeing this change, unless Github is just being dumb??

[the remark this answers is not in the document]
Comment (adilio) [reply README.md]: Apologies, was attempting to push a commit but VS Code isn't liking my auth. Pushed commit via GUI below.

**Review by steviecoaster — APPROVED** [review]

**Review by ryanrichter94 — COMMENTED** [review]
LGTM 🚀

**Review by ryanrichter94 — APPROVED** [review]

**Pull request merged**